add method to restart unbound on interface change

b4ldr opened the pull request:
pull request to implement #166

xaque208 commented on manifests/init.pp:
```diff
@@ -186,6 +186,14 @@
+      notify  => Exec['/etc/init/.d/unbound restart'],
+      content => "# Used by puppet-unbound\n${interface.join('\n')}",
+    }
+    exec {'/etc/init/.d/unbound restart':
```
This restart command should be tunable I think.

b4ldr commented:
@xaque208 agreed and updated, i think this default should work for bsd, ubuntu and debian

xaque208 commented:
Hmm, this isn't quite right.  I'm guessing its probably `systemctl` most everywhere on linux by now, no?  OpenBSD would be `/usr/sbin/rcctl restart unbound` and FreeBSD would be `/usr/sbin/service unbound restart`  I can add those before commit if helpful.

b4ldr commented:
both ubuntu and redhat use service as a wrapper top either systemctl, /etc/init.d or what ever upstart uses.  ill add something in for openbsd

b4ldr commented:
i have added exceptions for openbsd and freebsd.  I have also moved to using systemctl by default as it looks like the `service` command is a back-port hack which is likely gonna be removed in the future.  Further i changed the default of redhat 6 and ubuntu 14.04 to use the service command because i know that didn't have system.d.  i suspect there will need to some more exceptions but i dontknow them

xaque208 commented:
Runs great in my environment.  Thanks @b4ldr.
